src/main/scala/io/github/paoloboni/http/HttpClient.scala

```diff
@@ -100,8 +97,8 @@ sealed class HttpClient[F[_]: Async: Client: LogWriter](implicit
         }
       }
       handled <- decoded.fold(
-        decodingFailure => E.raiseError(decodingFailure),
-        response => F.pure(response)
+        decodingFailure => implicitly[Async[F]].raiseError(decodingFailure),
```

Comment:
Can you change this to
```suggestion
        decodingFailure => Async[F].raiseError(decodingFailure),
```